_posts/2019-06-25-dynamic_linear_models_tfprobability/dynamic_linear_models_tfprobability.Rmd

```diff
@@ -0,0 +1,805 @@
+    affiliation: RStudio
+    affiliation_url: https://www.rstudio.com/
+slug: keydana2019dlmtfp
+bibliography: bibliography.bib
+date: 06-24-2019
+categories:
+  - Probability and statistics 
+  - Time Series
+output:
+  distill::distill_article:
+    self_contained: false
+preview: images/thumb.png
+---
+
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE, eval = FALSE)
+```
+
+Welcome to the world of _state space models_. ^[For an excellent yet concise introduction to state space models, see chapter 18 of Murphy (2012) [@Murphy-2012].] There is a _latent process_, hidden from our eyes; and there are _observations_ we make about the things it produces. The process evolves due to some hidden logic (_transition model_); and the way it produces the observations follows some hidden logic (_observation model_). There is noise in process evolution, and there is noise in observation. If the transition and observation models both are linear, and process as well as observation noise are Gaussian, we have a _linear-Gaussian state space model_ (SSM). The task is to infer the latent state from the observations. The most famous technique is the _Kálmán filter_.
+
+In practical applications, two characteristics of SSMs are especially attractive. 
+
+For one, they let us estimate dynamically changing parameters. In regression, the parameters can be viewed as a hidden state; we may thus have a slope and an intercept that vary over time. When parameters can vary, we speak of _dynamic linear models_ (DLMs). This is the term we'll use throughout this post when referring to this class of models.
+
+Second, linear-Gaussian SSMs are useful in time series forecasting because Gaussian processes can be _added_. A time series can thus be framed as, e.g., the sum of a linear trend and a process that varies seasonally.
+
+Using [tfprobability](https://rstudio.github.io/tfprobability/), the R wrapper to TensorFlow Probability, we illustrate both aspects here. Our first example will be on _dynamic linear regression_. In a detailed walkthrough, we show on how to fit such a model, how to obtain filtered, as well as smoothed, estimates of the coefficients, and how to obtain forecasts. 
+Our second example then illustrates process additivity. This example will build on the first and may also serve as a quick recap of the overall procedure.
+
+Let's jump in. 
+
+
+## Dynamic linear regression example: Capital Asset Pricing Model (CAPM)
+
+Our code builds on the recently released versions of TensorFlow and TensorFlow Probability: 1.14 and 0.7, respectively. 
+
+```{r}
+devtools::install_github("rstudio/tfprobability")
+
+# also installs TensorFlow Probability v. 0.7
+tensorflow::install_tensorflow(version = "1.14")
+
+library(tensorflow)
+library(tfprobability)
+
+library(tidyverse)
+library(zeallot)
+```
+
+
+Note how there's one thing we used to do lately that we're _not_ doing here: We're not enabling eager execution. We say why in a minute.
```

Comment:
I think you can simplify this to "Note that we're _not_ enabling eager execution. We say why in a minute."